spec/features/users_can_edit_a_budget_spec.rb

```diff
@@ -84,7 +84,72 @@
       expect(historical_event.report).to eq(report)
     end
 
+    scenario "the revision history of a budget can be viewed" do
```

Comment:
This is quite long, I don't think it's too hard to follow, but happy to try to split it out somehow if needed.